_posts/2018-08-XX-custom-resources-in-0.5.0.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title:  "Strimzi adopts Custom Resources"
+date: 2018-08-XX
+author: tom bentley`
+---
+
+Strimzi is all about making it easy to run Kafka on a Kubernetes or OpenShift cluster.
+
+Until Strimzi 0.5.0 we've being using `ConfigMaps` as the way for Strimzi users to express their desired cluster state.
+That had a number of pros and cons, which we'll go into in a moment, but the main thing people upgrading to 0.5.0 will notice is that we are now using Custom Resources.
+
+<!--more-->
+
+# What is a Custom Resource anyway?
+
+A [Custom Resource](TODO) is a way for someone to extend a Kubernetes cluster with a new resource kind.
+They are often used with the [operator pattern](TODO) as a way of describing the desired resource which the operator will read and act on.
+Because Strimzi uses the operator pattern to make it easy to run Kafka on Kubernetes and OpenShift it's a natural fit for the project to make use of this Kubernetes feature.
+
+# What are the benefits of using Custom Resources?
+
+There are several benefits. Having a more natural, expressive and hierarchical way to represent the desired cluster means that the resources Strimzi users have to write are more readable and a much better fit with native Kubernetes or OpenShift resources such as `Deployments`.
+
+Related to that, the Kubernetes apiserver can perform validation when custom resources get created or modified. This means silly mistakes in the specification of the user's Kafka cluster are caught sooner and reported at the time the resource is created or modified. With ConfigMaps there was very limited validation and users were left to inspect the Cluster Operator logs to find out why their cluster didn't deploy as they expected.
+
+A third benefit of using Custom Resources is that they fit better with [Role Based Access Control](TODO) (RBAC).
```

Comment:
Yet again ... TODO ... you get the idea for the other TODOs as well I guess, so I will stop repeating this comment :-D